src/main/scala/mesosphere/marathon/metrics/SlidingAverageSnapshot.scala

```diff
@@ -0,0 +1,133 @@
+package mesosphere.marathon
+package metrics
+
+import java.time.Duration
+import java.util.logging.Logger
+
+import akka.Done
+import akka.actor.{ Actor, ActorRefFactory, Props }
+import kamon.Kamon
+import kamon.metric.{ Entity, EntitySnapshot }
+import com.typesafe.config.Config
+import kamon.metric.SubscriptionsDispatcher.TickMetricSnapshot
+import kamon.metric.instrument.CollectionContext
+import kamon.util.{ MapMerge, MilliTimestamp }
+import mesosphere.marathon.metrics.Metrics.subscribe
+
+/**
+  * Calculates sliding average entity snapshot
+  */
+class SlidingAverageSnapshot(val averagingWindow: Duration) {
+  private val logger: Logger = Logger.getLogger(getClass.getName)
+
+  val collectionContext: CollectionContext = Kamon.metrics.buildDefaultCollectionContext
+
+  /**
+    * The current summarised snapshot
+    */
+  private var currentSnapshot: TickMetricSnapshot = TickMetricSnapshot(MilliTimestamp.now, MilliTimestamp.now, Map.empty)
+
+  /**
+    * A ring buffer that collects the snapshots
+    */
+  private var averageRing: Array[TickMetricSnapshot] = ringFactory(Kamon.config)
```

Comment:
How about [`org.apache.commons.collections.buffer.CircularFifoBuffer`](https://commons.apache.org/proper/commons-collections/javadocs/api-3.2.2/org/apache/commons/collections/buffer/CircularFifoBuffer.html).

Reply:
I will definitely have a look 👍

However I wanted something quite simple. (Note that the `ringFactory` is a porting artifact from my Kamon `1.0` implementation. There the config can change during the execution and I was using the factory to create the new ring).

Reply:
You can still have the factory but I'd us an existing implementation.

Reply:
Ok, after our discussion I think it's fine to use the custom ring buffer. Querying the first and last element is easier in your solution.